funcx_sdk/funcx/serialize/facade.py

```diff
@@ -25,9 +45,57 @@ def __init__(self):
         for key in METHODS_MAP_DATA:
             self.methods_for_data[key] = METHODS_MAP_DATA[key]()
 
+    def _start_off_process_checker(self):
+        """ Kicks off off_process_checker using subprocess and returns the port on which it
+        is listening.
+        """
+        std_file = f'/tmp/{uuid.uuid4()}'
+        logger.debug(f"Writing off_process_checker logs to: {std_file}")
+        std_out = open(std_file, 'w')
+        self.serialize_proc = subprocess.Popen(['off_process_checker.py'], stdout=std_out, stderr=subprocess.STDOUT)
+        with open(std_file) as f:
+            for i in range(200, 3000, 200):
+                time.sleep(i / 100)  # Sleep incrementally waiting for process start
+                for line in f.readlines():
+                    if 'BINDING TO' in line:
+                        _, port = line.strip().split(':')
+                        port = int(port)
+                        return port
+
+        raise Exception("Failed to determine off_process_checker's port")
+
     def _list_methods(self):
         return self.methods_for_code, self.methods_for_data
 
+    def cleanup(self):
+        logger.debug("Cleaning up")
+        if self.use_offprocess_checker:
+            self.off_proc_client.close()
+            self.serialize_proc.terminate()
+
+    def deserialize_check(self, serialized_msg, timeout=1):
+        """
+        Parameters
+        ----------
+        serialized_msg
+
+        timeout : int
+           timeout in seconds to wait for result
+        """
+        if not self.use_offprocess_checker:
+            return False
+
+        with self.serialize_lock:
```

Comment:
Can you explain why we need a multiprocessing lock here? Just curious

Reply:
If funcX were to be integrated into a multi-threaded system, we could see multiple threads trying to register functions. Sockets aren't threadsafe, so without a lock we could have a super annoying bug. This is just safer and very inexpensive.